155 Support us page

frigus02 opened the pull request:
[Trello - Support Pride Page](https://trello.com/c/7nGZf47s/155-support-pride-page)

![image](https://user-images.githubusercontent.com/3579251/39047822-3b240296-4493-11e8-85cc-26e1a3877859.png)

It also addresses #133 for some screens.

## Test plan

* Go to the "Support us" tab.
* Do all kinds of weird shit on the page. It should not break.
* "Donate" and "Sponsor" buttons should open blank pages in the app.
* "Volunteer" and "Buy merchandise" buttons should open the browser on example.com.

## Pre-flight check-list

Before raising a pull request

* [x] Documentation
* [x] Unit tests

## Pre-merge check-list

* [x] Link to [Trello ticket](https://trello.com/c/7nGZf47s/155-support-pride-page)
* ~Any risky areas identified~
* [x] Test plan provided
* [ ] Tester approved

## Tester check-list

* [ ] Tested on multiple devices / OS versions (Test on the physical device(s) you have access to, otherwise use BrowserStack):

  * [x] Galaxy S8 (Android 7.0/8.0)
  * [x] Galaxy S7 (Android 6.0)
  * [ ] iPhone X (iOS 11.x)
  * [ ] iPhone 8 (11.x)
  * [ ] iPhone 7 (iOS 10.x)

  Optional:

  * [ ] Google Pixel 2
  * [ ] Galaxy S8 Plus
  * [ ] Galaxy S7 Edge
  * [ ] Galaxy S6
  * [x] iPhone 7+
  * [x] iPhone 6

Accessibility Testing (If applicable)

* [x] Text-to-Voice (Android: Voice Assistant / iOS: Speak Selection)
* [x] Large Text (=<200%)
* [ ] Landscape Screen orientation

Weak connection testing (If applicable)

* [ ] Airplane mode
* [ ] 2G/3G Network Settings

frigus02 commented:
This is still missing the URLs for "Volunteer" and "Buy merchandise" links. Also it is missing the shapes in the background of the 4 buttons.

But since this is the entry page to the Donate and Sponor pages, I think it makes sense to merge it like this and add the missing pieces later.

RGBboy commented on src/screens/SupportUsScreen/index.js:
```diff
@@ -0,0 +1,115 @@
+  container: {
+    flex: 1
+  },
+  scrollContainer: {
```
The `backgroundColor` should be `whiteColor` if I am reading the design correctly. It looks slightly grey to me which I think is coming from the parent view.

RGBboy commented on src/screens/SupportUsScreen/SupportUsSectionHeader.js:
```diff
@@ -0,0 +1,36 @@
+    color: lightNavyBlueColor,
+    margin: 16,
+    paddingHorizontal: 8,
+    lineHeight: 40
```
On iOS this looks a bit top heavy. I suggest using:

```
paddingTop: 4,
lineHeight: 36
```

Existing (with descender text):
<img width="310" alt="40" src="https://user-images.githubusercontent.com/1566539/38799026-326a7936-415b-11e8-8449-1fed1aeee336.png">

With change (with descender text):
<img width="310" alt="36-4" src="https://user-images.githubusercontent.com/1566539/38799048-3e20447c-415b-11e8-8e8e-2e8e0296cc2d.png">

frigus02 replied:
Nice. Will try this. Text alignment seems to differ a bit between Android and iOS. So I will need to find a good middle gound, I think.

JunBadger commented:
- Is the background currently a grey? It looks off-white (and should be #fff)
- Is the left margin between the text and left hand side of CTA 20px? Looks slightly bigger in build
- Cell will upload the shapes for you @frigus02

frigus02 commented:
Yeah, the background is gray. I will fix this and check the margin. It should be 20px, though.

frigus02 commented:
* I fixed the background color.
* I aligned the text title text better (thanks @RGBboy, your suggestion works great)
* I fixed the padding of the CTA buttons.
* I added the background shapes.

There 2 remaining issues:
* I needed to give the shapes a general `borderRadius` instead of a radius only on the left/right side, because it's the only thing iOS supports.
* When the user resizes the font, it will increase the height of the CTA buttons. But the shapes don't resize with it.

Someone can pick this up during the week. Or I can have a look at this on friday.

cellyn-tan commented:
Downloaded the build re: background shapes on my phone (iPhone X) - looks like the shapes are not contained within the CT for  **Buy Merchandise** and **Partner with us**

![img_7917](https://user-images.githubusercontent.com/38290023/38920304-daaa4848-42ea-11e8-9c3e-280959f1baaf.PNG)

frigus02 commented:
Weird. The reason is:
> When the user resizes the font, it will increase the height of the CTA buttons. But the shapes don't resize with it.

But I wonder what height the buttons have. Did you scale the font down (make it smaller)?

RGBboy commented:
@frigus02  What if we instead set an image as the background image? I would suggest that we just create larger images that look good when the button grows vertically. They don't need to scale, they just reveal more as the button grows.

frigus02 commented:
Background image: Can work, I think. The donate button needs 2 background images, though. One for the left and one for the right side.

Larger images, which just reveal more as the button grows is a good idea. :+1:

RGBboy commented:
Apologies, I just saw that React Native does not support background images. I still think there is value in design providing bigger images if the button grows. I think you may be able to mitigate the overflow problem by setting the overflow on the Image to `hidden`

cellyn-tan commented:
@frigus02 @RGBboy Ah got it. The font size I set on my phone was slightly smaller than the default font size. How about if we set a minimum height for the CTAs?

frigus02 commented:
React Native has basic support for background images using the `<ImageBackground>` component. We could maybe nest 2 of those:

```jsx
<ImageBackground source={bgLeft}>
  <ImageBackground source={bgRight}>
    Donate
  </ImageBackground>
</ImageBackground>
```

We need to see how we can align the images then. Might be worth a try.

frigus02 commented:
I think the overflow solution sounds good for the images. I tried to just scale the images and it didn't look that bad, either 🙂

declanslevin commented:
There's a small typo for the Donate button:
<img width=400 src="https://user-images.githubusercontent.com/33623842/39055087-35a1f932-44ab-11e8-9ed0-a61d1f34cdd1.jpg">

It should say "Every donation keeps us marching"
